src/macros.rs

```diff
@@ -0,0 +1,50 @@
+macro_rules! implement_register {
```

Comment:
Is this module declared or used anywhere? It seems you forgot to delete this code from "lib.rs".

Reply:
Sorry, I've force pushed a bad revision.
Changed in new push.